src/index.ts

```diff
@@ -64,8 +64,58 @@ export default function (schema: Schema, pluginOptions?: IPluginOptions) {
         return await createFindPromise(this, options, _query, _projection).explain(verbose);
     }
 
+    function createAggregatePromise<T>(
+        mongoCollection: Model<T>,
+        options: IPaginateOptions,
+        pipeline: PipelineStage[],
+    ) {
+        // Determine sort and limit for pagination
+        const sort = generateSort(options);
+
+        const defaultLimit = (pluginOptions && pluginOptions.defaultLimit ? pluginOptions.defaultLimit : 10);
+        const useDefaultLimit = isNaN(options.limit) || options.limit < 0 || options.limit === 0 && pluginOptions && pluginOptions.dontAllowUnlimitedResults;
+        const unlimited = options.limit === 0 && (!pluginOptions || !pluginOptions.dontAllowUnlimitedResults);
+        options.limit = useDefaultLimit ? defaultLimit : options.limit;
+    
+        // Apply pagination to the pipeline
+        const paginatedPipeline = [...generateAggregatePipeline(options), ...pipeline, { $sort: sort as any }];
+
+        if (!unlimited) {
+            paginatedPipeline.push({ $limit: options.limit + 1 });
+        }
+    
+        // Execute the aggregate query
+        const cursor = mongoCollection.aggregate<T>(paginatedPipeline);
+    
+        return cursor;
+    }
+    
+    async function aggregatePaged<T>(
```

Comment:
Looks good! Could you also update the README.md file?

Reply:
All right, updated